iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java

```diff
@@ -315,27 +315,48 @@ private boolean updateConfigNodeLeader(TSStatus status) {
     return false;
   }
 
-  @Override
-  public TSystemConfigurationResp getSystemConfiguration() throws TException {
+  /**
+   * The frame of execute RPC, include logic of retry and exception handling.
+   *
+   * @param call which rpc should call
+   * @param check check the rpc's result
+   * @return rpc's result
+   * @param <T> the type of rpc result
+   * @throws TException if fails more than RETRY_NUM times, throw TException(MSG_RECONNECTION_FAIL)
+   */
+  private <T> T executeRemoteCallWithRetry(Operation<T> call, Function<T, Boolean> check)
+      throws TException {
     for (int i = 0; i < RETRY_NUM; i++) {
       try {
-        TSystemConfigurationResp resp = client.getSystemConfiguration();
-        if (!updateConfigNodeLeader(resp.status)) {
-          return resp;
+        T result = call.execute();
+        if (!check.apply(result)) {
+          return result;
         }
       } catch (TException e) {
         logger.warn(
```

Comment:
maybe we should override `void warn(String var1, Throwable var2);`

Reply:
done